src/api/roadmap.ts

```diff
@@ -15,3 +15,30 @@ export const getSearchList = async (searchText: string) => {
   );
   return response;
 };
+
+export const getLevelList = async () => {
+  const response = await axios.get(
+    `${ENDPOINTS.ROADMAPS}levels`,
+  );
+  return response;
+};
+
+export const getDomainList = async () => {
+  const response = await axios.get(
+    `${ENDPOINTS.ROADMAPS}domains`,
+  );
+  return response;
+};
+
+export const getFiltredListWithSearch = async (obj:{search: string, level:string, domain:string, category:string}, page:number, pageSize: number = 6) => {
+  const response = await axios.get(
+    `${ENDPOINTS.ROADMAPS}search?page=${page}&page_size=${pageSize}&search=${obj.search}${obj.level !== ''? `&entry_level=${obj.level}`:''}${obj.domain !== ''? `&meta_domain=${obj.domain}`:''}${obj.category !== ''? `&local_domain=${obj.category}`:''}`,
+  );
```

Comment:
I think you'd better to separate ${obj.level !== ''? `&entry_level=${obj.level}`:''}${obj.domain !== ''? `&meta_domain=${obj.domain}`:''}${obj.category !== ''? `&local_domain=${obj.category}`:''}` filtering part in single parts and include it in path string, may it'll be more clear

Reply:
Stanislav, may you answer whether it is necessary to break it down into constants, or is it already clear?

Reply:
To answer this question you just need to take a quick look at this code. Are you all understand what's going on?
Fo me a little difficult understand what will be with query string)

1. you have mistake in function name - `Filtered`
2. obj can be anything, change name for something like filters or params...
3. you need put this obj type i general types and use it in query and in component because they are the same( don't write big types in params)
4. better save queries in string and then put in get url, it's really more clear. You can do it yourself or use [query-string](https://www.npmjs.com/package/query-string)